Detections/AzureActivity/AzDiagSettingsDeleted.yaml

```diff
@@ -19,17 +19,20 @@ relevantTechniques:
   - T1562.008
 query: |
   AzureActivity
-    | where OperationNameValue == 'MICROSOFT.INSIGHTS/DIAGNOSTICSETTINGS/DELETE' and ActivityStatusValue == "Start"
-    | extend ParentResource_a = split(_ResourceId,"/")
-    | extend ParentResourceLength = array_length(ParentResource_a)-4
-    | extend ParentResourceSplit = array_split(ParentResource_a,ParentResourceLength)
-    | extend resource = strcat_array(ParentResourceSplit[0],"/")
-    | project Diagdelete = TimeGenerated, Caller, ResourceProviderValue, _ResourceId, SubscriptionId, ResourceGroup, OperationNameValue, ActivityStatusValue, ActivitySubstatusValue, Start=TimeGenerated, resource, CallerIpAddress
-    | join kind=leftanti  ( AzureActivity
-    | where OperationNameValue != 'MICROSOFT.INSIGHTS/DIAGNOSTICSETTINGS/DELETE' and OperationNameValue endswith "/DELETE"
-    | where ActivityStatusValue == 'Start'
-    | project Caller, ResourceProviderValue, resource = tostring(parse_json(Properties).resource), SubscriptionId, ResourceGroup, OperationNameValue, ActivityStatusValue, ActivitySubstatusValue, Start=TimeGenerated) on $left.resource == $right.resource
-    | project Caller, ResourceProviderValue, resource, SubscriptionId, ResourceGroup, OperationNameValue, ActivityStatusValue, ActivitySubstatusValue, Start, CallerIpAddress
+  | where OperationNameValue =~ "MICROSOFT.INSIGHTS/DIAGNOSTICSETTINGS/DELETE"
+  | summarize
+      TimeGenerated = arg_max(TimeGenerated, Properties),
+      ActivityStatusValue = make_list(ActivityStatusValue),
```

Comment:
Can we use make_set instead of make_list here? Also, please add a max size parameter

Reply:
Done